Fix taxes totals mismatch at reports table

tomalec opened the pull request:
:warning: This PR requires https://github.com/woocommerce/woocommerce/pull/51270 to get the correct items count

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Before this PR, the tax reports table was using the total number of tax codes from `reports/taxes/stats` endpoint which counts distinct tax codes. This could lead to confusing results, as that number may not match the number of items in the reports table. The reports table list all tax-code + tax-value pairs, as they may change over time.

This PR renders both values  to provide complete information

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Place an order with a tax, let's say `PL 22% TAX-PL 1`
3. Go to `/wp-admin/admin.php?page=wc-settings&tab=tax&section=standard`
4. Change the value of that tax `PL 18% TAX-PL 1`, 
5. Place another order with updated tax
6. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Ftaxes`
7. Select a period that covers the orders


#### Before
![image](https://github.com/user-attachments/assets/a99bfdb1-a3a1-4aed-b381-e7f560eda2f6)


#### After

![image](https://github.com/user-attachments/assets/01076f6d-0231-41dd-ab95-1e95705e265f)


<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

tomalec committed:
Add group_by clause for total_results calculation

Fixes
- https://github.com/woocommerce/woocommerce/issues/48841
- https://github.com/woocommerce/woocommerce/issues/49635

tomalec committed:
Add changelog entry

tomalec committed:
Use both values

tomalec committed:
Default total taxcodes to 0

tomalec committed:
Render the correct number of totaly results in pagingation summary.

tomalec committed:
Merge branch 'trunk' into fix/49635-48841-refunded-taxes

tomalec committed:
Merge branch 'fix/49635-48841-refunded-taxes' into fix/49635-tax-totals

tomalec commented:
A seemingly unrelated E2E failed. I updated branches and re-run CI. If it still fails, I will continue the investigation tomorrow.

eason9487 commented on plugins/woocommerce-admin/client/analytics/report/taxes/table.js:
```diff
@@ -143,7 +143,16 @@ class TaxesReportTable extends Component {
 		return [
 			{
-				label: _n( 'tax code', 'tax codes', taxesCodes, 'woocommerce' ),
+				label: _n( 'tax', 'taxes', taxesCodes, 'woocommerce' ),
```
Should it use `totalResults` instead of `taxesCodes`?

Placing two orders with the same tax name but different rates can get this result:

![image](https://github.com/user-attachments/assets/c7c025e4-f937-4769-800b-aca17c524eaa)

eason9487 reviewed (commented):
Thanks for the fix. It works as expected, but the parameter used for getting singular or plural string may need to be adjusted and a changelog needs to be added.

tomalec committed:
add changelog entry

tomalec committed:
Fix singular/plural param

Address https://github.com/woocommerce/woocommerce/pull/51218#discussion_r1756264685

tomalec committed:
Merge branch 'trunk' into fix/49635-48841-refunded-taxes

tomalec committed:
Merge branch 'fix/49635-48841-refunded-taxes' into fix/49635-tax-totals

tomalec commented:
Thanks, @eason9487, for the review. I addressed your comments. Would you mind another look?

eason9487 reviewed (approved):
LGTM

tomalec committed:
Merge branch 'trunk' into fix/49635-tax-totals
